pdns/slavecommunicator.cc

```diff
@@ -509,6 +509,19 @@ void CommunicatorClass::addSlaveCheckRequest(const DomainInfo& di, const ComboAd
   Lock l(&d_lock);
   DomainInfo ours = di;
   ours.backend = 0;
+  string remote_address = remote.toString();
+
+  // When adding a check, if the remote addr from which notification was
+  // received is a master, clear all other masters so we can be sure the
+  // query goes to that one.
+  for (vector<string>::iterator it = ours.masters.begin(); it != ours.masters.end(); ++it) {
```

Comment:
Please use C++ 2011 for loops `for(const auto& master : ours.masters) { if (master == remote_address) { ... } }`